core/src/main/scala/com/raphtory/algorithms/temporal/Ancestors.scala

```diff
@@ -47,8 +47,18 @@ class Ancestors(seed: String, time: Long, delta: Long = Long.MaxValue, directed:
         if (vertex.name() == seed) {
           val edges = (if (directed) vertex.getInEdges() else vertex.getEdges())
             .filter(e => e.earliestActivity().time < time)
-            .filter(e => e.lastActivityBefore(time).time > time - delta)
-          edges.foreach(e => vertex.messageVertex(e.ID(), e.lastActivityBefore(time).time))
+            .filter(e =>
+              e.lastActivityBefore(time) match {
+                case Some(event) => event.time > time - delta
+                case None        => false
+              }
+            )
```

Comment:
No more need for the double filtering now? The second filter will handle the case where `e.earliestActivity().time > time` correctly now.